src/core/inverted_index_reader.rs

```diff
@@ -206,10 +206,50 @@ impl InvertedIndexReader {
         self.termdict.get_async(term.serialized_value_bytes()).await
     }
 
-    /// Returns a block postings given a `Term`.
+    async fn get_term_range_async(
+        &self,
+        terms: impl std::ops::RangeBounds<Term>,
+        limit: Option<u64>,
+    ) -> io::Result<impl Iterator<Item = TermInfo> + '_> {
+        use std::ops::Bound;
+        let range_builder = self.termdict.range();
+        let range_builder = match terms.start_bound() {
+            Bound::Included(bound) => range_builder.ge(bound.serialized_value_bytes()),
+            Bound::Excluded(bound) => range_builder.gt(bound.serialized_value_bytes()),
+            Bound::Unbounded => range_builder,
+        };
+        let range_builder = match terms.end_bound() {
+            Bound::Included(bound) => range_builder.le(bound.serialized_value_bytes()),
+            Bound::Excluded(bound) => range_builder.lt(bound.serialized_value_bytes()),
+            Bound::Unbounded => range_builder,
+        };
+        let range_builder = if let Some(limit) = limit {
+            range_builder.limit(limit)
+        } else {
+            range_builder
+        };
+
+        let mut stream = range_builder.into_stream_async().await?;
+
+        let iter = std::iter::from_fn(move || stream.next().map(|(_k, v)| v.clone()));
+
+        // limit on stream is only an optimization to load less data, the stream may still return
+        // more than limit elements.
+        let mut limit = limit;
+        let iter = iter.take_while(move |_e| match limit {
```

Comment:
maybe this should just be `.take(limit.unwrap_or(usize::MAX))`, even on 32b, limiting a range to "just" 4 billion terms shouldn't be an issue (or rather, there would probably be other problems before)

Reply:
I'd rather have that yes.

Reply:
yeah. let's go with the simpler code

Reply:
Can you make the change to use Iterator::take

Reply:
done